modules/albo-installation.adoc

```diff
@@ -53,236 +52,173 @@ $ cat <<EOF > "${SCRATCH}/trust-policy.json"
-  channel: stable-v1.0
-  installPlanApproval: Automatic
+  channel: stable-v1
   name: aws-load-balancer-operator
   source: redhat-operators
   sourceNamespace: openshift-marketplace
-  startingCSV: aws-load-balancer-operator.v1.0.0
+  config:
+    env:
+    - name: ROLEARN
+      value: "${OPERATOR_ROLE_ARN}"
 EOF
 ----
-+
-. Deploy an instance of the AWS Load Balancer Controller using the Operator:
-+
-[NOTE]
-====
-If you get an error here wait a minute and try again, it means the Operator has not completed installing yet.
-====
-+
-[source,terminal]
-----
-$ cat << EOF | oc apply -f -
-apiVersion: networking.olm.openshift.io/v1
-kind: AWSLoadBalancerController
-metadata:
-  name: cluster
-spec:
-  credentials:
-    name: aws-load-balancer-operator
-EOF
-----
-+
-. Check the that the Operator and controller pods are both running:
-+
-[source,terminal]
-----
-$ oc -n aws-load-balancer-operator get pods
-----
-+
-You should see the following, if not wait a moment and retry:
-+
-[source,terminal]
-----
-NAME                                                             READY   STATUS    RESTARTS   AGE
-aws-load-balancer-controller-cluster-6ddf658785-pdp5d            1/1     Running   0          99s
-aws-load-balancer-operator-controller-manager-577d9ffcb9-w6zqn   2/2     Running   0          2m4s
-----
 
-[id="aws-load-balancer-operator-validating-the-deployment_{context}"]
-== Validating the deployment
-
-. Create a new project:
+. Create an AWS IAM policy for the AWS Load Balancer Controller.
+.. Download the appropriate IAM policy:
 +
 [source,terminal]
 ----
-$ oc new-project hello-world
+$ curl -o ${SCRATCH}/controller-permission-policy.json https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/v2.12.0/docs/install/iam_policy.json
 ----
-+
-. Deploy a hello world application:
+.. Create the permission policy for the Controller:
 +
 [source,terminal]
 ----
-$ oc new-app -n hello-world --image=docker.io/openshift/hello-openshift
+$ aws iam create-policy \
+    --region "$REGION" \
+    --policy-name aws-load-balancer-controller-policy \
+    --policy-document file://${SCRATCH}controller-permission-policy.json
 ----
 +
-. Configure a NodePort service for the AWS ALB to connect to:
+Take note of the Controller policy ARN in the output. This is referred to as the `$CONTROLLER_POLICY_ARN` for the remainder of this process.
+
+. Create an AWS IAM role for the AWS Load Balancer Controller:
+.. Create the trust policy for the Controller role:
 +
 [source,terminal]
 ----
-$ cat << EOF | oc apply -f -
-apiVersion: v1
-kind: Service
-metadata:
-  name: hello-openshift-nodeport
-  namespace: hello-world
-spec:
-  ports:
-    - port: 80
-      targetPort: 8080
-      protocol: TCP
-  type: NodePort
-  selector:
-    deployment: hello-openshift
+$ cat <<EOF > controller-trust-policy.json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+          "Federated": "${OIDC_ARN}"
```

Comment:
OIDC_ARN has not been set in this document. To keep consistency set it the same way as above. Correct line:

`"Federated":  "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_ENDPOINT}"`